tools/fv_grid_tools.F90

```diff
@@ -1874,30 +2039,136 @@ subroutine setup_aligned_nest(Atm)
+         if( is_master() ) then
 
-         call mpp_recv(p_grid( isg-ng:ieg+1+ng, jsg-ng:jeg+1+ng,1:2), size(p_grid( isg-ng:ieg+1+ng, jsg-ng:jeg+1+ng,1:2)), &
-                       Atm%parent_grid%pelist(1))
-      endif
+            call mpp_recv(p_grid( isg-ng:ieg+1+ng, jsg-ng:jeg+1+ng,1:2), size(p_grid( isg-ng:ieg+1+ng, jsg-ng:jeg+1+ng,1:2)), &
+                 Atm%parent_grid%pelist(1))
+
+         endif
+
+         call mpp_broadcast( p_grid(isg-ng:ieg+ng+1, jsg-ng:jeg+ng+1, :), &
+              (ieg-isg+2+2*ng)*(jeg-jsg+2+2*ng)*ndims, mpp_root_pe() )
 
-      call mpp_broadcast( p_grid(isg-ng:ieg+ng+1, jsg-ng:jeg+ng+1, :), &
-           (ieg-isg+2+2*ng)*(jeg-jsg+2+2*ng)*ndims, mpp_root_pe() )
+       end if
+
+       if (use_timer) call mpp_clock_end (id_timer1)
+       if (use_timer) call mpp_clock_begin (id_timer2)
+
+    !!  Setup full grid for nest; confusingly called grid_global.  Each nest PE is computing the grid lat/lons for the entire nest
+    !!    not just its section.
+    !!  INPUTS:  ioffset, joffset, p_grid
+    !!  OUTPUTS:  grid_global
+
+      ! Begin calculate shifted version of global_grid
+
+      if (first_time) allocate(shift_p_ind(1-ng:npx  +ng,1-ng:npy  +ng,4))   ! TODO need to deallocate this somewhere
+
+      if (.not. first_time) then
+
+         ! Make copies of grid_global and p_ind to validate that code is correct
+         allocate( out_grid( lbound(grid_global,1):ubound(grid_global,1), &
+              lbound(grid_global,2):ubound(grid_global,2), &
+              lbound(grid_global,3):ubound(grid_global,3), &
+              lbound(grid_global,4):ubound(grid_global,4) ) )
+
+         if (debug_log) print '("[INFO] WDR bounds grid_global setup_nest_grid npe=",I0," grid_global(",I0,"-",I0,",",I0,"-",I0,",",I0,"-",I0,",",I0,"-",I0,")")', this_pe, lbound(grid_global,1), ubound(grid_global,1), &
+              lbound(grid_global,2), ubound(grid_global,2), &
+              lbound(grid_global,3), ubound(grid_global,3), &
+              lbound(grid_global,4), ubound(grid_global,4)
+
+         if (debug_log) print '("[INFO] WDR bounds out_grid setup_nest_grid npe=",I0," out_grid(",I0,"-",I0,",",I0,"-",I0,",",I0,"-",I0,",",I0,"-",I0,")")', this_pe, lbound(out_grid,1), ubound(out_grid,1), &
+              lbound(out_grid,2), ubound(out_grid,2), &
+              lbound(out_grid,3), ubound(out_grid,3), &
+              lbound(out_grid,4), ubound(out_grid,4)
+
+         out_grid = grid_global
+
+         if ( delta_i_c .ne. 0 ) then
+            if (debug_log) print '("[INFO] setup_nest_grid EOSHIFT delta_i_c=",I0," start. npe=",I0)', delta_i_c, this_pe
+            out_grid = eoshift(out_grid, refinement * delta_i_c, 0.0, 1)
+         end if
+
+         if (delta_j_c .ne.  0) then
+            if (debug_log) print '("[INFO] setup_nest_grid EOSHIFT delta_j_c=",I0," start. npe=",I0)', delta_j_c, this_pe
+            out_grid = eoshift(out_grid, refinement * delta_j_c, 0.0, 2)
+         end if
```

Comment:
Syntax error when using GNU compiler:
```
/home/dusan/ufs/src/model/FV3/atmos_cubed_sphere/tools/fv_grid_tools.F90:2127:65:

 2127 |             out_grid = eoshift(out_grid, refinement * delta_i_c, 0.0, 1)
      |                                                                 1
Error: ‘boundary’ argument of ‘eoshift’ intrinsic at (1) must be the same type and kind as ‘array’
/home/dusan/ufs/src/model/FV3/atmos_cubed_sphere/tools/fv_grid_tools.F90:2132:65:

 2132 |             out_grid = eoshift(out_grid, refinement * delta_j_c, 0.0, 2)
      |                                                                 1
Error: ‘boundary’ argument of ‘eoshift’ intrinsic at (1) must be the same type and kind as ‘array’
make[2]: *** [FV3/atmos_cubed_sphere/CMakeFiles/fv3.dir/build.make:517: FV3/atmos_cubed_sphere/CMakeFiles/fv3.dir/tools/fv_grid_tools.F90.o] Error 1
```
Fix:
```diff
diff --git a/tools/fv_grid_tools.F90 b/tools/fv_grid_tools.F90
index 596a75e..dd32778 100644
--- a/tools/fv_grid_tools.F90
+++ b/tools/fv_grid_tools.F90
@@ -2124,12 +2124,12 @@ contains
 
          if ( delta_i_c .ne. 0 ) then
             if (debug_log) print '("[INFO] setup_nest_grid EOSHIFT delta_i_c=",I0," start. npe=",I0)', delta_i_c, this_pe
-            out_grid = eoshift(out_grid, refinement * delta_i_c, 0.0, 1)
+            out_grid = eoshift(out_grid, refinement * delta_i_c, 0.0_R_GRID, 1)
          end if
 
          if (delta_j_c .ne.  0) then
             if (debug_log) print '("[INFO] setup_nest_grid EOSHIFT delta_j_c=",I0," start. npe=",I0)', delta_j_c, this_pe
-            out_grid = eoshift(out_grid, refinement * delta_j_c, 0.0, 2)
+            out_grid = eoshift(out_grid, refinement * delta_j_c, 0.0_R_GRID, 2)
          end if
 
          shift_p_ind(:,:,1) = shift_p_ind(:,:,1) + delta_i_c
```

Reply:
I am going to fix this in a different way -- the third argument is a fill value, and defaults to the type-appropriate value of zero for any numeric input arrays.  So I will restructure these calls to remove that argument and explicitly name the final DIM argument.

Reply:
I have committed the change to remove the BOUNDARY argument, which causes it to use the default value, which is the correct single or double precision value of 0.  This will provide identical functionality as the initial code, but will be portable to various compilers.